sources/mongodb_pipeline.py

```diff
@@ -0,0 +1,95 @@
+from typing import List
+
+import dlt
+from dlt.common import pendulum
+
+from mongodb import mongodb, mongodb_collection
+
+def load_select_collection_from_database() -> None:
+    """Use the mongodb source to reflect an entire database schema and load select tables from it.
+
+    This example sources data from a sample mongo database data from [mongodb-sample-dataset](https://github.com/neelabalan/mongodb-sample-dataset).
+    """
+    # Create a pipeline
+    pipeline = dlt.pipeline(
+        pipeline_name="pipeline1", destination="duckdb", dataset_name="mongo_data"
+    )
+
+    # Credentials for the sample database.
+    # Note: It is recommended to configure credentials in `.dlt/secrets.toml` under `sources.sql_database.credentials`
+    connection_url = "mongodb://admin:123456@localhost:27017"
+    
+    database = "sample_mflix"
+    # Configure the source to load a few select collections incrementally
+    mflix = mongodb(connection_url, database=database, incremental=dlt.sources.incremental("date")).with_resources("comments")
+
+    # Run the pipeline. The merge write disposition merges existing rows in the destination by primary key
+    info = pipeline.run(mflix, write_disposition="merge")
+    print(info)
+
+    database = "sample_analytics"
+    # Load some other tables with replace write disposition. This overwrites the existing tables in destination
+    analytics = mongodb(connection_url, database=database).with_resources("transactions", "accounts")
+    info = pipeline.run(analytics, write_disposition="replace")
+    print(info)
+
+    database = "sample_airbnb"
+    # Load a table incrementally with append write disposition
+    # this is good when a table only has new rows inserted, but not updated
+    airbnb = mongodb(connection_url, database=database).with_resources("listingsAndReviews")
+    # TODO: Not working
+    airbnb.listingsAndReviews.apply_hints(incremental=dlt.sources.incremental("last_scraped"))
```

Comment:
look at sql_database example
```
from typing import List

import dlt
from dlt.sources.credentials import ConnectionStringCredentials
from dlt.common.time import ensure_pendulum_datetime
from dlt.common import pendulum

from sql_database import sql_database, sql_table


def load_select_tables_from_database() -> None:
    """Use the sql_database source to reflect an entire database schema and load select tables from it.

    This example sources data from the public Rfam MySQL database.
    """
    # Create a pipeline
    pipeline = dlt.pipeline(
        pipeline_name="rfam", destination="postgres", dataset_name="rfam_data", full_refresh=True
    )

    # Credentials for the sample database.
    # Note: It is recommended to configure credentials in `.dlt/secrets.toml` under `sources.sql_database.credentials`
    credentials = ConnectionStringCredentials(
        "mysql+pymysql://rfamro@mysql-rfam-public.ebi.ac.uk:4497/Rfam"
    )

    # Configure the source to load a few select tables incrementally
    source_1 = sql_database(credentials).with_resources("clan")
    # Add incremental config to the resources. "updated" is a timestamp column in these tables that gets used as a cursor
    # source_1.family.apply_hints(incremental=dlt.sources.incremental("updated"))
    source_1.clan.apply_hints(incremental=dlt.sources.incremental("updated", initial_value=ensure_pendulum_datetime("2014-06-02 15:19:28+00:00")))

    # Run the pipeline. The merge write disposition merges existing rows in the destination by primary key
    info = pipeline.run(source_1, write_disposition="append")
    print(info)
```
this works here: pulls just record after the date in `initial_value`

Reply:
IDK what "not working" means... most common problems:
- defining the cursor on non existing column (will raise exception with explanation)
- using `initial_value` with type incompatible with what is in cursor column - exception on musing max function

if by not working you mean that it is not filtering data then let's talk on slack

Reply:
I found that the incremental is applied with the `apply_hints` but when extracting the data, the incremental is removed by the function [`apply_hint_args`](https://github.com/dlt-hub/dlt/blob/devel/dlt/pipeline/pipeline.py#L818) from dlt. I could not figure out yet if I am passing something wrong or what, will keep looking at it.

Reply:
Let me also check. maybe you've found some kind of a bug